nodeJS/express/views.md

```diff
@@ -0,0 +1,165 @@
+
+We use template engines to create our views. More specifically, template engines are used to create template files that transform the template into HTML. Any variables defined in our template files are replaced with actual data. Additionally we can insert logic into our template file, eg. render the user's username once they have logged in. This would not be possible with plain HTML.
+
+In this course, we will use [EJS](link: https://ejs.co/) which lets us write in JavaScript to create HTML markup. `EJS`' syntax is very similar to that of HTML, meaning that the learning curve is relatively low compared to other template engines.
+
+### Lesson overview
+
+This section contains a general overview of topics that you will learn in this lesson.
+
+- EJS Installation
+- EJS Syntax
+- Application of EJS
+
+### Setting up EJS
+
+Let's get started with EJS! Install EJS into your application by typing the following into your terminal:
+
+```bash
+npm install ejs
+```
+
+At the root of your project, create a subfolder called `views`.
+
+Next, we need to let our app know that we intend to use `EJS` as a template engine, as well as where to look for view files.
+
+In your `app.js` file, add the following:
+
+```js
+app.set("views", path, join(__dirname, "views"));
+app.set("view engine", "ejs");
+```
+
+This enables `EJS` as the view engine, and that our app should look for templates in the `/views` subdirectory.
+
+### EJS syntax
+
+In `EJS`, the `<%` and `%>` tags allow us to use JavaScript. Whatever is in these tags will behave like JavaScript. This lets us write conditional statements, `for` loops, as well as use variables.
+
+In order to work with variables, we use the `<%=` tag. This lets us convert variables into values when the application is loaded. Otherwise, stick with `<%` when using JavaScript logic.
+
+Here's a quick example that makes includes arrays and loop logic.
+
+```js
+<% const animals = ["Cat", "Dog", "Lemur", "Hawk"] %>
+
+<% animals.map(animal => { %>
+- <%= animal%>s are cute <% }) %>
+```
+
+### Using EJS with Express
+
+Let's use `EJS` with `Express`. First, create an `EJS` template file called `index.ejs`, and add the following:
+
+```html
+<html>
+  <body>
+    <%= message %>
+  </body>
+</html>
+```
+
+And in app.js, render this template file in one of your routes:
+
+```js
+app.get("/", (req, res) => {
+  res.render("index", { message: "EJS rocks!" });
+});
+```
+
+When the `"/"` route is matched, the template file matching `"index"` is rendered with the message value passed, such that you should see:
+
+```
+EJS rocks!
+```
+
+### Reusable web components
```

Comment:
I'm not sure if calling it 'web components' is the right term. Maybe "Reusable templates?
```suggestion
### Reusable templates
```

Reply:
Reusable templates is probably better. I chose web components, as I was only thinking about components such as headers, footers, sidebar, etc. But I guess those are only specific applications of templates, so ultimately templates would be the better word to use.